tonic/src/codec/json.rs

```diff
@@ -0,0 +1,76 @@
+use std::marker::PhantomData;
+use bytes::{Buf, BufMut};
+use prost::bytes::{Buf, BufMut};
+use crate::codec::{Codec, DecodeBuf, Decoder, EncodeBuf, Encoder};
+use crate::Status;
+
+extern crate serde;
+extern crate serde_derive;
+
+pub struct JsonEncoder<T>(PhantomData<T>);
+
+impl<T: serde::Serialize> Encoder for JsonEncoder<T> {
+    type Item = T;
+    type Error = Status;
+
+    fn encode(&mut self, item: Self::Item, buf: &mut EncodeBuf<'_>) -> Result<(), Self::Error> {
+        let s = serde_json::to_string(&item).expect("Message only errors if not enough space");
+        buf.put(s.as_bytes());
+
+        Ok(())
+    }
+}
+
+pub struct JsonDecoder<U>(PhantomData<U>);
+
+impl<U: for<'a> serde::Deserialize<'a>> Decoder for JsonDecoder<U> {
```

Comment:
Can we use this https://docs.rs/serde/latest/serde/de/trait.DeserializeOwned.html should be the same just a bit cleaner